File: Sexta_Generacion_CPCFI/clases/dia11/1633.cpp
#include <bits/stdc++.h>
using namespace std;
typedef long long ll;
#define TAM 1000000
#define MOD 1000000007

int main() {
    vector<ll> dp(TAM,0);
    ll n = 1;
    cin >> n;
    dp[0]=1;dp[1]=1;dp[2]=2;dp[3]=4;dp[4]=8;dp[5]=16;dp[6]=32;
    
    for(ll i = 7; i <= n; i++) {
        for(int j = 1; j <= 6; j++) {
            dp[i] += dp[i-j];
            dp[i] %= MOD;
        }
    }
    
    cout << dp[n] << endl;
}

File: Sexta_Generacion_CPCFI/clases/dia11/1633.py
# cses
MOD = 1000000007
TAM = 1000000

n = int(input())
dp = [0 for _ in range(TAM+1)]
dp[0],dp[1],dp[2],dp[3],dp[4],dp[5],dp[6] = 1,1,2,4,8,16,32

for i in range(7,n+1):
    for j in range(1,7):
        dp[i] += dp[i-j]
        dp[i] %= MOD
print(dp[n])
